### PR TITLE
tests: Moving tests around

### PR DESCRIPTION
**Pull request by mithro**
* Moves architecture specific stuff under each directory.
 * Puts all the makefiles in one common place. 
 * Makes the common tests work for each architecture.

### CODE REVIEW

**Comment by mithro** [comment]
@elmsfu PTAL

File: ice40/make/tests.mk [hunk 1]
```diff
@@ -61,16 +61,19 @@ endif
 # ---------------------------------------------
 # ---------------------------------------------
 
-INPUT_IO_FILE=$(TEST_DIR)/$(BOARD).pcf
+INPUT_IO_FILE=$(wildcard $(TEST_DIR)/$(BOARD).pcf)
```
Comment (elmsfu) [inline ice40/make/tests.mk]: It's not clear when we'd need wildcard. Is there a case when a * would be in the BOARD?

File: ice40/tests/iceff/Makefile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ARCH=ice40
+DEVICE=HX1K
```
Comment (elmsfu) [inline ice40/tests/iceff/Makefile]: DEVICE=hx1k
Needs to be lowercase now I believe

File: ice40/tests/iceff/Makefile [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+include ../../../make/tests.mk
```
Comment (elmsfu) [inline ice40/tests/iceff/Makefile]: Is PACKAGE required?

File: ice40/make/tests.mk (hunk 1 above)
Comment (mithro) [reply ice40/make/tests.mk]: `$(wildcard xxx)` will return empty when the file doesn't exist.

Comment (elmsfu) [reply ice40/make/tests.mk]: I think this is clearer
`INPUT_IO_FILE=$(if $BOARD,$(TEST_DIR)/$(BOARD).pcf)`

File: tests/Makefile [hunk 4]
```diff
@@ -1,6 +1,7 @@
 
 SUBDIRS=$(sort $(wildcard [0-9]-*))
-ARCHS=$(sort $(basename $(notdir $(wildcard make/arch/*.mk))))
+ARCHS=$(sort $(basename $(notdir $(wildcard ../*/make/tests.mk))))
+ARCHS=$(sort $(patsubst ../%/make/tests.mk,%,$(wildcard ../*/make/tests.mk)))
```
[the remark this answers is not in the document]
Comment (mithro) [reply tests/Makefile]: Done.

File: ice40/make/tests.mk [hunk 5]
```diff
@@ -61,16 +61,19 @@ endif
...
 endif
 ifeq ($(PACKAGE),)
-$(error No $$PACKAGE set.)
+#$(error No $$PACKAGE set.)
+PACKAGE=qn84
 endif
 ifeq ($(PROG_TOOL),)
-$(error No $$PROG_TOOL set.)
+#$(error No $$PROG_TOOL set.)
+PROG_TOOL=true
```
[the remark this answers is not in the document]
Comment (mithro) [reply ice40/make/tests.mk]: Reverted.

File: ice40/tests/iceff/Makefile (hunk 2 above)
Comment (mithro) [reply ice40/tests/iceff/Makefile]: Done.

File: ice40/tests/iceff/Makefile (hunk 3 above)
Comment (mithro) [reply ice40/tests/iceff/Makefile]: Fixed.

**Pull request merged**